components/app-proxy/vm.tf

```
@@ -6,78 +6,65 @@ module "tags" {
   builtFrom   = var.builtFrom
 }
 
-resource "azurerm_resource_group" "this" {
+resource "azurerm_resource_group" "app_proxy_rg" {
```

https://www.terraform-best-practices.com/naming#resource-and-data-source-arguments

resources names should not contain resource type.
no real benefit of naming this as it's the only one of the type.

```suggestion
resource "azurerm_resource_group" "this" {
```